lib/kvdb/kvdb_rest.c

```diff
@@ -97,6 +101,260 @@ rest_kvdb_get(
+}
+
+static merr_t
+rest_kvs_param_get(
+    const char *      path,
+    struct conn_info *info,
+    const char *      url,
+    struct kv_iter *  iter,
+    void *            context)
+{
+    merr_t          err = 0;
+    struct hse_kvs *kvs = context;
+
+    /* Check from single parameter or all parameters */
+    if (strcmp(path, url)) {
+        size_t      needed_sz;
+        char *      tmp;
+        size_t      buf_sz = 128;
+        const char *param = path + strlen(url) + 1; /* move past the final '/' */
+        char *      buf = malloc(buf_sz);
+
+        if (!buf)
+            return merr(ENOMEM);
+
+        err = ikvdb_kvs_param_get(kvs, param, buf, buf_sz, &needed_sz);
+        if (needed_sz >= buf_sz && !err) {
+            buf_sz = needed_sz + 1;
+            tmp = realloc(buf, buf_sz);
+            if (!tmp)
+                return merr(ENOMEM);
+
+            buf = tmp;
+
+            err = ikvdb_kvs_param_get(kvs, param, buf, buf_sz, NULL);
+        }
+
+        if (!err && write(info->resp_fd, buf, strnlen(buf, buf_sz)) == -1)
+            err = merr(errno);
+
+        free(buf);
+    } else {
+        char * str;
+        cJSON *merged, *cp_json, *rp_json;
+
+        cp_json = kvs_cparams_to_json(((struct kvdb_kvs *)kvs)->kk_cparams);
+        if (!cp_json)
+            return merr(ENOMEM);
+
+        rp_json = kvs_rparams_to_json(&((struct kvdb_kvs *)kvs)->kk_ikvs->ikv_rp);
+        if (!rp_json) {
+            cJSON_Delete(cp_json);
+            return merr(ENOMEM);
+        }
+
+        merged = cJSONUtils_MergePatchCaseSensitive(cp_json, rp_json);
+        assert(merged);
+
+        str = cJSON_PrintUnformatted(merged);
+        cJSON_Delete(merged);
+        if (!str) {
+            return merr(ENOMEM);
+        }
+
+        if (!err && write(info->resp_fd, str, strlen(str)) == -1)
+            err = merr(errno);
+
+        cJSON_free(str);
+    }
+
+    return err;
+}
+
+static merr_t
+rest_kvs_param_put(
+    const char *      path,
+    struct conn_info *info,
+    const char *      url,
+    struct kv_iter *  iter,
+    void *            context)
+{
+    const char *     param;
+    struct kvdb_kvs *kvs = context;
+    const bool       has_param = strcmp(path, url);
+
+    /* Check for case when no parameter is specified, /kvdb/0/kvs/1/params */
+    if (!has_param)
+        return merr(EINVAL);
+
+    param = path + strlen(url) + 1;
+
+    return kvs_rparams_set(&kvs->kk_ikvs->ikv_rp, param, info->data);
+}
+
+static merr_t
+rest_kvdb_mclass_info_get(
+    const char *      path,
+    struct conn_info *info,
+    const char *      url,
+    struct kv_iter *  iter,
+    void *            context)
+{
+    merr_t                 err = 0;
+    enum mpool_mclass      mclass = MP_MED_INVALID;
+    struct hse_mclass_info mc_info;
+    cJSON *                root;
+    char *                 str;
+
+    for (size_t i = 0; i < NELEM(mpool_mclass_to_string); i++) {
```

Comment:
Looping from MP_MED_BASE to MP_MED_COUNT won't work here?

Reply:
Can do